### PR TITLE
Homemade polehammers

### PR DESCRIPTION
**Pull request by SirPendrak**
#### Summary
Content "Add two new earlygame polehammers"

#### Purpose of change

More weapon crafting choices

#### Describe the solution

Add two new weapons and crafting recipies for them, their idea is weponising standard hammer.
Survivor polehammer (changed to homemade polehammer) is roughly equivalent to spear_knife_propper.
Makeshift survivor polehammer (changed to makeshift homemade polehammer) is roughly equivalent of spear_knife_superior.
Overall, they are a bit easier to make, a tiny bit worse as a weapons and have different tehniques.

#### Describe alternatives you've considered

Adding bronze hammer variants. Adding variant that has been sharpened with angle grinder.
I was wondering if theese should have "hooking weaponary" similar to lucerne hammer, but decided that claw hammer head might be too small to reliable use some advanced martialarts.
EDIT: Messed a bit IRL with hammers. They definitely can be used to hook.

#### Testing

Crafted them and used aganist zeds. They have medicore damage compared to lategame reach bashing weapons, but are faster, thus make kiting easier, and have better stamina to dmg ratio. Overall, solid weapons, especially when confronted with trivial crafting recepies.

#### Additional context

I hope I understand term "survivor weapon" in the right way. Its not super weapon, but clever usage of existing components to create simple and functional desing.

### CODE REVIEW

File: data/json/items/melee/bludgeons.json [hunk 1]
```diff
@@ -1171,6 +1171,42 @@
...
+  {
+    "id": "survivor_polehammer_makeshift",
+    "copy-from": "survivor_polehammer",
+    "type": "GENERIC",
+    "name": { "str": "makeshift survivor polehammer" },
+    "description": "A standard hammerhead mounted at the end of a long stick.  It is far from being a proper weapon, but it allows you to bash zombie heads from a safe distance.",
+    "delete": { "flags": [ "DURABLE_MELEE" ] },
+    "weight": "1800 g",
+    "volume": "1650 ml",
+    "to_hit": { "grip": "solid", "length": "long", "surface": "any", "balance": "uneven" },
```
Comment (kevingranade) [inline data/json/items/melee/bludgeons.json:1205]: Pretty minor nit, surface isn't any, you want to hit square and aligned with the striking face.  You have a 2 square inch or so impact zone at the end of a meter long pole that needs to make contact,  that's not easy.

**Comment by Maddremor** [comment]
I suggest using another name than "survivor polehammer". The survivor prefix is very ambiguous and tells the player nothing about what the item is.

**Comment by SirPendrak** [comment]
> I suggest using another name than "survivor polehammer". The survivor prefix is very ambiguous and tells the player nothing about what the item is.

Well, I oppened a synonyms dictionary, and there arent any good words. Best I can think of is homemade polehammer and makeshift homemade polehammer

File: data/json/items/melee/bludgeons.json (hunk 1 above)
Comment (SirPendrak) [reply data/json/items/melee/bludgeons.json:1205]: > Pretty minor nit, surface isn't any, you want to hit square and aligned with the striking face. You have a 2 square inch or so impact zone at the end of a meter long pole that needs to make contact, that's not easy.

Well, you're probably right, I just based this on lucerne hammer. And that weapon is a mess. It has surface "any", even if its hammerhead striking area is not much bigger than standard hammer. IRL their shape is ofter even more reliant on good alighment. And lucerne hammer has pierce damage, probably from beak or spearhead, but they benefit from mentioned "any" surface.

Back to topic, giving these polehammers "point" surface will be huge nerf, then they gonna need increase damage to be similar of metal plated quaterstaves or something

**Comment by Kantonine** [comment]
YMMV, but i feel that it could be better described using "simple makeshift polehammer" and "makeshift polehammer" for both the weak and durable versions respectively.

**Comment by Uwuewsky** [comment]
The division into simple and makeshift is a little unintuitive imo; at one time I thought simple was better, although it’s the opposite. `simple makeshift` is a bit wordy.

It's a little difficult to find an adjective for crap weapon and even crappier weapon, so that they don't sound like interchangeable synonyms, and so that the crafting progression was clear to the player.

It would be nice to have some kind of standard on this matter. Like `crude`/`primitive`/`improvised` for `simple makeshift` and `simple`/`basic`/`makeshift` for current `makeshift`.

**Comment by Maddremor** [comment]
To take a step back: What is the difference between the two items?

**Comment by SirPendrak** [comment]
> To take a step back: What is the difference between the two items?

You can click on commit to see the code. Or click "file changed". Makeshift one is made with long stick and Hammer. Has worse grip, worse ballance, no durable melee. Proper one uses hammer and true spearshaft.

**Comment by Maddremor** [comment]
I didn't ask about the implementation, but the conceptual difference. Seeing as you are attaching something to be swung sideways, would the simpler construction even stay intact? There doesn't necessarily have to be two polehammers just because there are two knife spears.

File: data/json/items/melee/bludgeons.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/items/melee/bludgeons.json:1205]: Make the change to make it correct, then we can use the DPS tests to determine whether to make other changes to damage/etc.

File: data/json/recipes/weapon/bashing.json [hunk 2]
```diff
@@ -189,6 +189,33 @@
...
+    "result": "survivor_polehammer_makeshift",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_BASHING",
+    "skill_used": "fabrication",
+    "time": "15 m",
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 1 } ],
+    "proficiencies": [ { "proficiency": "prof_carving", "time_multiplier": 1.5, "skill_penalty": 0.15 } ],
+    "components": [ [ [ "stick_long", 1 ], [ "long_pole", 1 ] ], [ [ "hammer", 1 ] ] ]
+  },
```
Comment (anothersimulacrum) [inline data/json/recipes/weapon/bashing.json:218]: How does this recipe work? What are you doing? Is it removing the head from the hammer, carving an appropriate place to mount it, and sticking the head on? How are you doing that in 15-20 minutes?

File: data/json/recipes/weapon/bashing.json [hunk 3]
```diff
@@ -189,6 +189,33 @@
...
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_BASHING",
+    "skill_used": "fabrication",
+    "difficulty": 1,
+    "time": "30 m",
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 2 }, { "id": "HAMMER", "level": 1 } ],
+    "proficiencies": [ { "proficiency": "prof_carving", "time_multiplier": 1.5, "skill_penalty": 0.15 } ],
+    "components": [ [ [ "spear_shaft", 1 ] ], [ [ "hammer", 1 ], [ "survivor_polehammer_makeshift", 1 ] ], [ [ "nails", 3, "LIST" ] ] ]
+  },
```
Comment (anothersimulacrum) [inline data/json/recipes/weapon/bashing.json:205]: Then how do nails make that more stable? I don't think this justifies durable_melee particularly, and FRAGILE_MELEE may be justified for the lesser version.

File: data/json/items/melee/bludgeons.json (hunk 1 above)
Comment (anothersimulacrum) [reply data/json/items/melee/bludgeons.json:1205]: Make the change to make it correct, then we can use the DPS tests to determine what changes to make.

**Comment by SirPendrak** [comment]
> I didn't ask about the implementation, but the conceptual difference. Seeing as you are attaching something to be swung sideways, would the simpler construction even stay intact? There doesn't necessarily have to be two polehammers just because there are two knife spears.

Concept is that one version is moderatelly crappy weapon that uses premade tool as base that can be crafted in first hours of cataclysm and second version would be done in few first days of cataclysm using two durable components. As for durability for simpler one, long stick is big, bulky, and is used to craft proper shafts for other true polehammers, so from "fragile melee" and medium durability, i have chosen second.

File: data/json/recipes/weapon/bashing.json (hunk 2 above)
Comment (SirPendrak) [reply data/json/recipes/weapon/bashing.json:218]: Exactly what you described. Removing hammerhead can be done in 2 minutes. For carving place to mount it, you only need to carve like 3 cm lenght, and not much in diameter

File: data/json/recipes/weapon/bashing.json (hunk 3 above)
Comment (SirPendrak) [reply data/json/recipes/weapon/bashing.json:205]: Its not nails that make the most difference, but they are often used in mounting hammerheads and axeheads. The real difference comes from propper spearshaft, the same that is used in crafting other, weapon grade polearms. Best possible wooden handle + baisicaly indestructible but small hammerhead = durable weapon

Comment (anothersimulacrum) [reply data/json/recipes/weapon/bashing.json:205]: My understanding is that mounting a head with nails looks like this:
image
Which is not something you can do with a pre-fabricated hammer.

File: data/json/recipes/weapon/bashing.json (hunk 2 above)
Comment (anothersimulacrum) [reply data/json/recipes/weapon/bashing.json:218]: How are you removing a hammer head in two minutes with (at worst) a sharp rock? I'd expect at least hammering required for that.

Comment (SirPendrak) [reply data/json/recipes/weapon/bashing.json:218]: Fair enough, I will adjust it when I get in front of computer again

File: data/json/recipes/weapon/bashing.json (hunk 3 above)
Comment (SirPendrak) [reply data/json/recipes/weapon/bashing.json:205]: What I had in mind with nails was using them as wedges. I know most professional wedges for hammer and axe heads are small metal trianges, in CDDA that would probably translate to... scrap metal? Or small metal sheet. And that would be more confusing

Comment (TheShadowFerret) [reply data/json/recipes/weapon/bashing.json:205]: Small metal sheets + nuts & bolts instead of nails should be used for mounting then

**Comment by Maddremor** [comment]
> > I didn't ask about the implementation, but the conceptual difference. Seeing as you are attaching something to be swung sideways, would the simpler construction even stay intact? There doesn't necessarily have to be two polehammers just because there are two knife spears.
> 
> Concept is that one version is moderatelly crappy weapon that uses premade tool as base that can be crafted in first hours of cataclysm and second version would be done in few first days of cataclysm using two durable components. As for durability for simpler one, long stick is big, bulky, and is used to craft proper shafts for other true polehammers, so from "fragile melee" and medium durability, i have chosen second.

Okay, but how are you doing that? I don't see an easy way of attaching a hammerhead like [this one ](http://upload.wikimedia.org/wikipedia/commons/8/84/Claw-hammer.jpg) to a stick or shaft. What does the hammer equivalent to the tang of a knife look like?

**Comment by SirPendrak** [comment]
> > > I didn't ask about the implementation, but the conceptual difference. Seeing as you are attaching something to be swung sideways, would the simpler construction even stay intact? There doesn't necessarily have to be two polehammers just because there are two knife spears.
> > 
> > 
> > Concept is that one version is moderatelly crappy weapon that uses premade tool as base that can be crafted in first hours of cataclysm and second version would be done in few first days of cataclysm using two durable components. As for durability for simpler one, long stick is big, bulky, and is used to craft proper shafts for other true polehammers, so from "fragile melee" and medium durability, i have chosen second.
> 
> Okay, but how are you doing that? I don't see an easy way of attaching a hammerhead like [this one ](http://upload.wikimedia.org/wikipedia/commons/8/84/Claw-hammer.jpg) to a stick or shaft. What does the hammer equivalent to the tang of a knife look like?

In case of of more modernish hammer with polymer handle and no hole in hammerhead, this simple recipe is impossible. I think arround half of polymerhandle hammers have tang like the one on your picture. But in game we have hammers with wooden handle. And almost all of them have standard hole for mounting, often wider at the top, for more security fit with wedges. 
61-tYiWxOgL _AC_UF894,1000_QL80_

File: data/json/recipes/weapon/bashing.json (hunk 3 above)
Comment (SirPendrak) [reply data/json/recipes/weapon/bashing.json:205]: I dont understand how nuts and bolts can be used as wedges

**Comment by SirPendrak** [comment]
I had some problem with deciding weapon categories and proficiences.
For weapon to be able to use benefits of weapon proficiency, it needs to have weapon category defined ["HOOKING_WEAPONRY", "POLEARMS"], 
then you can learn proficiency in combat.
It also enables you to use martialarts with said weapon (Fior Di Battaglia, Silat etc.).
To train proficiency using practice crafting menu, the weapon needs to be included in requirements itemgroups. 
Requirements itemgroup for polearms states that it is only for "Any basic weapon you can use as a **cutting** polearm"
Polehammer is pure bashing. But it is polearm. Category is only for cutting polearms.
And then the recipe for training proficiency aslso trains **stabbing**...

**Comment by zachary-kaelan** [comment]
I would just call the makeshift one "hammer on a stick" and use duct tape or strings like the spike on a stick.

**Comment by SirPendrak** [comment]
> I would just call the makeshift one "hammer on a stick" and use duct tape or strings like the spike on a stick.

Why anybody would try to ducktape hammerhead when you can fit it at least somewhat decently with a little carving

**Comment by zachary-kaelan** [comment]
> Why anybody would try to ducktape hammerhead when you can fit it at least somewhat decently with a little carving

Someone with no fabrication skill isn't going to know how to do that, they're just going to duct tape it.

**Comment by kevingranade** [comment]
The main reason someone's not going to duct tape a hammer into a pole is it's not going to work. That's going to break very quickly, my only question is if it even survives one hit.

File: data/json/recipes/weapon/bashing.json [hunk 4]
```diff
@@ -189,6 +189,33 @@
...
+    "activity_level": "LIGHT_EXERCISE",
+    "result": "homemade_polehammer_makeshift",
+    "category": "CC_WEAPON",
+    "subcategory": "CSC_WEAPON_BASHING",
+    "skill_used": "fabrication",
+    "time": "30 m",
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 2 }, { "id": "HAMMER", "level": 1 } ],
+    "proficiencies": [ { "proficiency": "prof_carving", "time_multiplier": 1.5, "skill_penalty": 0.15 } ],
+    "components": [ [ [ "stick_long", 1 ], [ "long_pole", 1 ] ], [ [ "hammer", 1 ] ] ]
```
Comment (Maleclypse) [inline data/json/recipes/weapon/bashing.json:217]:
```suggestion
    "components": [ [ [ "stick_long", 1 ], [ "long_pole", 1 ] ], [ [ "hammer", 1 ] ], [ [ "superglue", 1 ] ] ]
```
